src/query/numeric_opfunc.c

```diff
@@ -3871,15 +3873,36 @@ numeric_db_value_print (const DB_VALUE * val, char *buf)
       /* Add the negative sign */
       if (temp[i] == '-')
 	{
-	  buf[nbuf] = '-';
-	  nbuf++;
+	  buf[nbuf++] = '-';
 	}
 
       /* Add decimal point */
       if (i == temp_size - scale)
 	{
-	  buf[nbuf] = '.';
-	  nbuf++;
+	  int k = temp_size - 1;
```

Comment:
This is an out of scope question.

Why is the size 80?
```
char temp[80];
```

I know this is because NUMERIC_MAX_STRING_SIZE is (80 + 1).
However, as a result of the test below, I don't know why 0 to 38 are needed.

Even when 38, the maximum value of scale, is specified, 38 repetitions must be passed meaninglessly.
If the scale is 5, 71 times are repeated meaninglessly.

e.g.
```
create table t1 (c1 numeric (38,0));
insert into t1 values (12345678901234567890123456789012345678.0); // Fail
insert into t1 values (12345678901234567890123456789012345678); // Success

select * from t1;

( '0', '0', '0', '0', '0', '0',    '0',    '0',    '0',    '0'   // 0~9
  '0', '0', '0', '0', '0', '0',    '0',    '0',    '0',    '0'   // 10~19
  '0', '0', '0', '0', '0', '0',    '0',    '0',    '0',    '0'   // 20~29
  '0', '0', '0', '0', '0', '0',    '0',    '0',    '1',    '2'   // 30~39
  '3', '4', '5', '6', '7', '8',    '9',    '0',    '1',    '2'   // 40~49
  '3', '4', '5', '6', '7', '8',    '9',    '0',    '1',    '2'   // 50~59
  '3', '4', '5', '6', '7', '8',    '9',    '0',    '1',    '2'   // 60~69
  '3', '4', '5', '6', '7', '8', '\000', '\000', '\000', '\000' ) // 70~79

create table t2 (c1 numeric (38,38));
insert into t2 values (0.12345678901234567890123456789012345678);

select * from t2;

( '0', '0', '0', '0', '0', '0',    '0',    '0',    '0',    '0'   // 0~9
  '0', '0', '0', '0', '0', '0',    '0',    '0',    '0',    '0'   // 10~19
  '0', '0', '0', '0', '0', '0',    '0',    '0',    '0',    '0'   // 20~29
  '0', '0', '0', '0', '0', '0',    '0',    '0',    '1',    '2'   // 30~39
  '3', '4', '5', '6', '7', '8',    '9',    '0',    '1',    '2'   // 40~49
  '3', '4', '5', '6', '7', '8',    '9',    '0',    '1',    '2'   // 50~59
  '3', '4', '5', '6', '7', '8',    '9',    '0',    '1',    '2'   // 60~69
  '3', '4', '5', '6', '7', '8', '\000', '\000', '\000', '\000' ) // 70~79
  
create table t3 (c1 numeric (38,5));
insert into t3 values (123456789012345678901234567890123.98765);

( '0', '0', '0', '0', '0', '0',    '0',    '0',    '0',    '0'   // 0~9
  '0', '0', '0', '0', '0', '0',    '0',    '0',    '0',    '0'   // 10~19
  '0', '0', '0', '0', '0', '0',    '0',    '0',    '0',    '0'   // 20~29
  '0', '0', '0', '0', '0', '0',    '0',    '0',    '1',    '2'   // 30~39
  '3', '4', '5', '6', '7', '8',    '9',    '0',    '1',    '2'   // 40~49
  '3', '4', '5', '6', '7', '8',    '9',    '0',    '1',    '2'   // 50~59
  '3', '4', '5', '6', '7', '8',    '9',    '0',    '1',    '2'   // 60~69
  '3', '9', '8', '7', '6', '5', '\000', '\000', '\000', '\000' ) // 70~79
```

Reply:
Wouldn't it be better to run the loop from the back?